src/sardana/taurus/core/tango/sardana/macro.py

```diff
@@ -1301,45 +1301,60 @@ def createMacroNode(macro_name, params_def, macro_params):
 
     if not new_interface:
         param_nodes = macro_node.params()
-        contain_param_repeat = False
         len_param_nodes = len(param_nodes)
-        for i, param_node in enumerate(param_nodes):
-            if isinstance(param_node, RepeatParamNode):
-                if contain_param_repeat:
-                    msg = "Only one repeat parameter is allowed"
-                    raise Exception(msg)
-                if i < len_param_nodes - 1:
-                    msg = "Repeat parameter must be the last one"
-                    raise Exception(msg)
-        # If ParamRepeat only one and as last parameter
-        # this ignores raw_parameters which exceeds the param_def
+        param_idx = 0
         for param_node, param_raw in zip(param_nodes, macro_params):
             if isinstance(param_node, SingleParamNode):
                 param_node.setValue(param_raw)
-            # the rest of the values are interpreted as repeat parameter
-            elif isinstance(param_node, RepeatParamNode):
+            # Repeat parameters that are not at the end.
+            elif (isinstance(param_node, RepeatParamNode) and
+                  param_idx < (len_param_nodes - 1)):
+                repeat_node = param_node.child(0)
+                # Add a new repeat node. This is needed when the raw values
+                # fill more repeat nodes that the minimum number of
+                # repetitions e.g. min=0.
+                if repeat_node is None:
+                    repeat_node = param_node.addRepeat()
+                if len(repeat_node.children()) > 1:
+                    msg = ("repeat parameter with more than one member must "
+                           "be defined at the end when intended to "
+                           "be used with spock syntax")
+                    raise Exception(msg)
+                member_node = repeat_node.child(0)
+                if isinstance(member_node, RepeatParamNode):
+                    msg = ("nested repeat parameters are not allowed when "
```

Comment:
I would suggest the following message instead:
`"Nested repeat parameters not allowed in (old) spock syntax. Use "bracket" syntax instead."`